src/js/components/PodDetail.js

```diff
@@ -0,0 +1,234 @@
+import {Confirm} from 'reactjs-components';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import Breadcrumbs from './Breadcrumbs';
+import InternalStorageMixin from '../mixins/InternalStorageMixin';
+import MarathonStore from '../stores/MarathonStore';
+import Service from '../structs/Service';
+import ServiceActionItem from '../constants/ServiceActionItem';
+import ServiceDestroyModal from './modals/ServiceDestroyModal';
+import ServiceDetailConfigurationTab from './ServiceDetailConfigurationTab';
+import ServiceDetailDebugTab from './ServiceDetailDebugTab';
+import PodDetailInstancesTab from './PodDetailInstancesTab';
+import ServiceFormModal from './modals/ServiceFormModal';
+import ServiceSuspendModal from './modals/ServiceSuspendModal';
+import PodInfo from './PodInfo';
+import TabsMixin from '../mixins/TabsMixin';
+import VolumeTable from './VolumeTable';
+
+const METHODS_TO_BIND = [
+  'closeDialog',
+  'onActionsItemSelection',
+  'onAcceptRestartConfirmDialog',
+  'onServiceDestroyModalClose',
+  'onServiceSuspendModalClose'
+];
+
+class PodDetail extends mixin(InternalStorageMixin, StoreMixin, TabsMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.tabs_tabs = {
+      instances: 'Instances',
+      configuration: 'Configuration',
+      debug: 'Debug'
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift(),
+      disabledDialog: null,
+      errorMsg: null,
+      serviceActionDialog: null
+    };
+
+    this.store_listeners = [
+      {
+        name: 'marathon',
+        events: [
+          'serviceRestartError',
+          'serviceRestartSuccess'
+        ]
+      }
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    if (this.props.mode === 'pod') {
```

Comment:
Are you doing anything here? And is there anything which you have to do specifically in the `componentWillMount` function which you can't do in `componentDidMount`?